cli/silence_import.go

```diff
@@ -59,20 +58,21 @@ func configureSilenceImportCmd(cc *kingpin.CmdClause) {
 	importCmd.Action(execWithTimeout(c.bulkImport))
 }
 
-func addSilenceWorker(ctx context.Context, sclient client.SilenceAPI, silencec <-chan *types.Silence, errc chan<- error) {
+func addSilenceWorker(ctx context.Context, sclient *silence.Client, silencec <-chan *models.PostableSilence, errc chan<- error) {
 	for s := range silencec {
-		silenceID, err := sclient.Set(ctx, *s)
-		sid := s.ID
+		params := silence.NewPostSilencesParams().WithContext(ctx)
+		params.Silence = s
+		postOk, err := sclient.PostSilences(params)
```

Comment:
This part should probably be adapted as `POST /api/v2/silences` can only return 200 or 400 and it should update (recreate) an existing silence.

https://github.com/prometheus/alertmanager/blob/f40ecee77326d169dec5092d28510c60a9593f0b/api/v2/openapi.yaml#L77-L86

Reply:
This seems to work similarly to v1 AFAICT.  In both cases if the silence ID does not exist, the API returns a 400 with "not found" in the error message.

Reply:
Digging more into the code I think we can refine the API here. The handler is returning any error from the silences as-is but it could intercept the `ErrNotFound` error and turn it into a 404 response. It would be cleaner for the client as it would only have to check the status code and not rely on the fact that the returned body contains the `not found` string. @mxinden @stuartnelson3 what do you think?

https://github.com/prometheus/alertmanager/blob/a94088406ab7212acadaa01f4e86467ee1eac8f6/api/v2/api.go#L660-L664

https://github.com/prometheus/alertmanager/blob/a94088406ab7212acadaa01f4e86467ee1eac8f6/silence/silence.go#L502-L511

Reply:
It would be nice to avoid the string comparison

Reply:
@simonpasquier sounds good to me.

Reply:
Created #1816 to add the 404 response to the v2 rest API.

Reply:
Rebased to get the changes from #1816 and updated the cli code to check the error type instead of doing the string compare.